workshop_sections/word2vec/README.md

```diff
@@ -1,120 +1,178 @@
+The `gcloud` CLI ships with a set of commands for interacting with the Google Cloud ML api `gcloud beta ml`. One of these commands is the local emulator, `gcloud beta ml local train` which runs your TensorFlow code locally in an environment which emulates that of Google Cloud ML workers. This can be an easy way to validate distributed code you plan to run in the Cloud.
 
- * `word_array`: A `np.ndarray` of words 
+You can run the following command *using either local or cloud data files* to train your model.
 
-NOTE: You may need to use a smaller slice depending on the amount of memory on your machine.
 ```
-targets, contexts = util.generate_batches(word_array[:-4000000])
+gcloud beta ml local train \
+    --package-path word2vec \
+    --module-name word2vec.task \
+    -- \
+    --output-path output/1 \
+    --vocab-file gs://tf-ml-workshop/word2vec/data/text8-vocab.tsv \
+    --train-data-file gs://tf-ml-workshop/word2vec/data/text8-train.pb2 \
+    --eval-data-file gs://tf-ml-workshop/word2vec/data/text8-eval.pb2
 ```
-(this may take 5-10 minutes)
 
-`word2vec.generate_batches` applies SkipGram processing to rolling windows in `word_indices`. The first element in the returned tuple is a vector of target word indices, while the second is a vector of the context word indices we'd like to predict.
+This will run your model in 5 processes (2 parameter servers and 2 workers). You will see output from all 5 in your terminal.
 
-### From TFRecords
+### Training in the Cloud with the Google Cloud Machine Learning API
 
-If you don't want to do preprocessing in memory, or you want to preprocess via another distributed graph computation framework, Tensorflow supports reading from a binary (protobuf) format called TFRecords. We have provided this file publicly available online. See below for how to train with this data.
+To train in the Cloud, we use `gcloud` to submit a training job. It's important to note that here we must use a Google Cloud Storage output directory. Make sure you have set up a Google Cloud Storage bucket and authorized the Google Cloud ML API to access it. See instructions [here](../../INSTALL.md#project-and-cloud-ml-setup).
 
-## Training the model
+After you have completed this, set `MY_BUCKET=gs://my-bucket` with your name, for the commands below.
 
-To define your model, run:
+Then submit a training job with:
 
 ```
-import tensorflow as tf
-import os
-import time
-workshop_root = os.path.dirname(os.path.dirname(os.getcwd()))
-output_dir = os.path.join(workshop_root, 'workshop-data', 'output', str(int(time.time())))
-tf.logging.set_verbosity(tf.logging.INFO)
-word2vec_model = tf.contrib.learn.Estimator(model_fn=word2vec.make_model_fn(index, counts.tolist()), model_dir=output_dir)
+gcloud beta ml jobs submit training myword2vectraining \
+  --staging-bucket $MY_BUCKET \
+  --module-name word2vec.task \
+  --package-path word2vec \
+  --config config.yaml \
+  -- \
+  --output-path $MY_BUCKET/word2vec/output/run1 \
```

Comment:
Yeah it will. I actually didn't want to require a new output-dir each time, so that they could see jobs are easily resumable if they crash.  I tried to make it clear with "run1" that you should increment every number but I'll call it out in prose as well.